### PR TITLE
Update pigeon_sort.py

### PR DESCRIPTION
**Pull request by wuyudi**
### **Describe your change:**

Update pigeon_sort.py

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by swapnanildutta — APPROVED** [review]

**Review by swapnanildutta — APPROVED** [review]

File: sorts/pigeon_sort.py [hunk 1]
```diff
@@ -26,29 +26,18 @@ def pigeon_sort(array):
...
-    # Manually finds the minimum and maximum of the array.
-    min = array[0]
-    max = array[0]
-
-    for i in range(len(array)):
-        if array[i] < min:
-            min = array[i]
-        elif array[i] > max:
-            max = array[i]
+    # Finds the minimum and maximum of the array.
```
Comment (mateuszz0000) [inline sorts/pigeon_sort.py:29]: Could we remove unnecessary comments? Code should be self documented

**Review by mateuszz0000 — CHANGES_REQUESTED** [review]

File: sorts/pigeon_sort.py [hunk 2]
```diff
@@ -26,7 +26,6 @@ def pigeon_sort(array):
     if len(array) == 0:
         return array
 
-    # Finds the minimum and maximum of the array.
     _min, _max = min(array), max(array)
 
     # Compute the variables
```
Comment (53jk1) [inline sorts/pigeon_sort.py:31]: unnecesary comment

**Review by 53jk1 — CHANGES_REQUESTED** [review]

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

**Review by xcodz-dot — APPROVED** [review]
@dhruvmanila check this out

**Review by dhruvmanila — APPROVED** [review]
LGTM!

**Pull request merged**